### PR TITLE
add isOpen prop that allows owners to override the isOpen state

### LINKED ISSUE #189: Ability to close dropdown list
Hi,

Thanks for all the hard work putting this together. One issue that I have is that I would like to be able to close the dropdown list when it is open, either at will or through a parameter that would close it when a selection is made.

Currently, if @props.mutli is true, the list stays open, requiring the user to click elsewhere in the UI in order to move on. While I understand the logic behind this, this behavior is not always desirable and it would be nice to have some control over it.

### PR DESCRIPTION
**Pull request by bobbyrenwick**
This closes #189.

You may want to change the name of the prop to something like `forceOpen` in order to disambiguate it from the state, but that's up to you!

### CODE REVIEW

**Comment by Paveltarno** [comment]
+1

File: src/Select.js [hunk 1]
```diff
@@ -122,7 +124,7 @@ const Select = React.createClass({
 			inputValue: '',
 			isFocused: false,
 			isLoading: false,
-			isOpen: false,
+			isOpen: this.props.isOpen !== null ? this.props.isOpen : false,
```
Comment (JKillian) [inline src/Select.js]: will this work properly if `isOpen` is `undefined`?

**Comment by JKillian** [comment]
+1 an `isOpen` prop would be great and would solve my issue in #962

File: src/Select.js (hunk 1 above)
Comment (bobbyrenwick) [inline src/Select.js]: Why would it be `undefined`, it defaults to `null`.

Comment (JKillian) [inline src/Select.js]: I guess you're right, I was thinking in the case of `<Select isOpen={undefined}/>`, but I suppose React uses the default value in this case?

Comment (bobbyrenwick) [inline src/Select.js]: I didn't know, so I made a [fiddle](https://jsfiddle.net/bobbyrenwick/a70Loco4/) that demonstrates what you suggested is the true behaviour!

**Comment by charlesbodman** [comment]
+1 need this badly. It's making it very difficult to style .

**Comment by ericbiewener** [comment]
@charlesbodman To force it open temporarily for styling, just set `isOpen` to `true` in the source code [here](https://github.com/JedWatson/react-select/blob/master/src/Select.js#L908).

**Comment by adidahiya** [comment]
bump @JedWatson, any word on this?

**Comment by dkh215** [comment]
Is this on ice?

**Comment by tonyjmnz** [comment]
This feature would be really useful for an use case I'm working on. I also think it would make the component much more flexible when paired with a flux architecture.

**Comment by vaclav-zeman** [comment]
Is this going to be implemented anytime soon?

**Comment by kajmagnus** [comment]
I'd just want to tell you about my use case: When React-Select opens, it hides the Save button in my dialog. So the user needs to understand that s/he has to either select an option, _or_ click somewhere outside React-Select to close it, for the Save button to become visible again. However, if s/he clicks outside the _dialog_, in order to close React-Select then the whole dialog closes and disappears.

This would likely confuse people (even I feel a bit confused). I think this can be solved, by leaving React-Select open always, and giving it enough space, so it won't hide the Save button.

(I don't want it to be "only" open — I want it to push the stuff below downwards too so it won't overlap the Save button. But this last thing seems to be a simple CSS style: `.Select-menu-outer { position: relative; }` (instead of `absolute`.) works fine for me.)

**Comment by ConneXNL** [comment]
Waiting for this one...

**Comment by urisavka** [comment]
Hey guys. Any news on this one? 
Or maybe is there any other way to force open dropdown from the start?

**Comment by tnrich** [comment]
@JedWatson  Please add this feature!

This PR is doing the same thing as https://github.com/JedWatson/react-select/pull/1015

**Comment by aleccool213** [comment]
bump

**Comment by thiagocoelho** [comment]
There is any fix/way to force the open state? I need to style the options but I can't inspect the elements to discovery what I need to change.

**Comment by MindRave** [comment]
Any news on this one? Would be a very useful prop indeed.

**Comment by haio** [comment]
+1 really need this feature!

**Comment by EnriqueVidal** [comment]
Is there still interest in this? I just pushed #1950, it works a little different though, rather than using `isOpen` it implements `alwaysOpen`, the test suite is green and I seem to be passing the coverage test, however I cannot run the example components via the `start` script, not sure if I'm missing anything in there.

Any advise/suggestion is more than welcomed.

**Comment by JedWatson** [comment]
Thanks for the PR @bobbyrenwick, I'm going to close it until we have more clarity on #1989

**Comment by leojacob7** [comment]
Why did you close this despite so many requests @JedWatson

**Pull request closed without merging**